File: backend/api/producer.go
package api

import (
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type Producer struct {
	Id          primitive.ObjectID `bson:"_id" json:"id,omitempty"`
	Name        string             `bson:"name" json:"name,omitempty"`
	Description string             `bson:"description" json:"description,omitempty"`
	Wkn         string             `bson:"wkn" json:"wkn,omitempty"`
	Isin        string             `bson:"isin" json:"isin,omitempty"`
	Symbol      string             `bson:"symbol" json:"symbol,omitempty"`
	IconId      string             `bson:"iconId" json:"iconId,omitempty"`
	Tendency    string             `bson:"tendency" json:"tendency,omitempty"`
	LastPrice   float64            `bson:"lastPrice" json:"lastPrice,omitempty"`
	PriceData   []PriceData        `bson:"priceData" json:"priceData,omitempty"`
}

File: App/app/src/main/java/com/example/balticbytebuccaneers/service/producer/Producer.kt
package com.example.balticbytebuccaneers.service.producer

data class Producer(
    val description: String?,
    val iconId: String?,
    val id: String?,
    val isin: String?,
    val lastPrice: Int?,
    val name: String?,
    val priceData: List<PriceData>?,
    val symbol: String?,
    val tendency: String?,
    val wkn: String?
)
